### PR TITLE
[tidb] Add all column type test for tidb.

### PR DESCRIPTION
**Pull request by GOODBOY008**
[tidb] Add all column type test for tidb.

### CODE REVIEW

File: flink-connector-tidb-cdc/src/main/java/com/ververica/cdc/connectors/tidb/table/utils/TiKVTypeUtils.java [hunk 1]
```diff
@@ -201,6 +244,10 @@ public static DataType getFlinkType(final org.tikv.common.types.DataType dataTyp
                     object = LocalTime.ofNanoOfDay(Long.parseLong(object.toString()));
                 }
                 break;
+            case "OffsetDateTime":
+                object =
```
Comment (leonardBang) [inline flink-connector-tidb-cdc/src/main/java/com/ververica/cdc/connectors/tidb/table/utils/TiKVTypeUtils.java:248]: `OffsetDateTime` should be mapped to `TIMESTAMP WITH TIMEZONE`, current implementation will loss the timezone information, we'd better use String here. BTW, `OffsetDateTime` can not be casted to `Timestamp`.

File: flink-connector-tidb-cdc/src/main/java/com/ververica/cdc/connectors/tidb/table/utils/TiKVTypeUtils.java [hunk 2]
```diff
@@ -233,17 +280,40 @@ public static DataType getFlinkType(final org.tikv.common.types.DataType dataTyp
...
+            case TypeTiny:
+                return Short.valueOf((short) Byte.toUnsignedInt(((Short) object).byteValue()));
+            case TypeShort:
+                return Integer.valueOf(Short.toUnsignedInt(((Integer) object).shortValue()));
+            case TypeInt24:
+                return Integer.valueOf(((int) object) & 0xffffff);
+            case TypeLong:
+                return Long.valueOf(Integer.toUnsignedLong(((Long) object).intValue()));
+            case TypeLonglong:
+                return DecimalData.fromBigDecimal(
```
Comment (leonardBang) [inline flink-connector-tidb-cdc/src/main/java/com/ververica/cdc/connectors/tidb/table/utils/TiKVTypeUtils.java:307]: Could you post the related tikv link here for let other developers know more?

**Review by leonardBang — COMMENTED** [review]
Thanks @GOODBOY008 for the great work, I left some comments

File: flink-connector-tidb-cdc/src/main/java/com/ververica/cdc/connectors/tidb/table/utils/TiKVTypeUtils.java (hunk 2 above)
Comment (GOODBOY008) [reply flink-connector-tidb-cdc/src/main/java/com/ververica/cdc/connectors/tidb/table/utils/TiKVTypeUtils.java:307]: Ok, I will add link https://docs.pingcap.com/zh/tidb/stable/data-type-numeric#%E5%AD%98%E5%82%A8%E7%A9%BA%E9%97%B4%E4%BB%A5%E5%8F%8A%E5%8F%96%E5%80%BC%E8%8C%83%E5%9B%B4 here.

Comment (leonardBang) [reply flink-connector-tidb-cdc/src/main/java/com/ververica/cdc/connectors/tidb/table/utils/TiKVTypeUtils.java:314]: English doc is more recommended, https://docs.pingcap.com/tidb/stable/data-type-numeric

**Review by leonardBang — APPROVED** [review]
Thanks @GOODBOY008 for the great work, these improvements help our code more clean and strong. +1

**Pull request closed without merging**